src/badge.jsx

```diff
@@ -0,0 +1,119 @@
+const ThemeManager = require('./styles/theme-manager');
+
+// Badge
+export default React.createClass({
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  getChildContext () {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+  propTypes: {
+    className: React.PropTypes.string,
+    primary: React.PropTypes.bool,
+    secondary: React.PropTypes.bool,
+    primaryText: React.PropTypes.string.isRequired,
+    color: React.PropTypes.string,
+    backgroundColor: React.PropTypes.string,
+    style: React.PropTypes.object,
+  },
+  getInitialState() {
+    return {
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+    };
+  },
+  componentWillReceiveProps(nextProps, nextContext) {
+    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+    this.setState({
+      muiTheme: newMuiTheme,
+    });
+  },
+  _getBackgroundColor() {
+    return this.props.backgroundColor
+      ? this.props.backgroundColor
+      : this.props.primary
+        ? this.getTheme().primaryColor
+        : this.props.secondary
+          ? this.getTheme().secondaryColor
+          : this.getTheme().color;
+  },
+  _getLabelColor() {
+    return this.props.labelColor
+      ? this.props.labelColor
+        : this.props.primary
+          ? this.getTheme().primaryTextColor
+          : this.props.secondary
+            ? this.getTheme().secondaryTextColor
+            : this.getTheme().textColor;
+  },
+  getThemeBadge() {
+    return this.state.muiTheme.badge;
+  },
+  getTheme() {
+    return this.state.muiTheme.raisedButton;
+  },
+  getStyles() {
+    const style = {
+      root: {
+        backgroundColor: 'none',
+        position: 'relative',
+        display: 'inline-block',
+        marginRight: '20px',
+      },
+      badge: {
+        display: '-webkit-box; display: -webkit-flex; display: flex',
+        WebkitBoxOrient: 'horizontal',
+        WebkitBoxDirection: 'normal',
+        WebkitFlexDirection: 'row',
+        msFlexDirection: 'row',
+        flexDirection: 'row',
+        WebkitFlexWrap: 'wrap',
+        msFlexWrap: 'wrap',
+        flexWrap: 'wrap',
+        WebkitBoxPack: 'center',
+        WebkitJustifyContent: 'center',
+        msFlexPack: 'center',
+        justifyContent: 'center',
+        WebkitAlignContent: 'center',
+        msFlexLinePack: 'center',
+        alignContent: 'center',
+        WebkitBoxAlign: 'center',
+        WebkitAlignItems: 'center',
+        msFlexAlign: 'center',
+        alignItems: 'center',
+        position: 'absolute',
+        top: '-11px',
+        right: '-24px',
+        fontWeight: Typography.fontWeightMedium,
+        fontSize: '12px',
+        width: '22px',
+        height: '22px',
+        borderRadius: '50%',
+        backgroundColor: this._getBackgroundColor(),
```

Comment:
Only called once, I would move all the function to get the style properties inside `getStyles`.